daemon/delete.go

```diff
@@ -71,9 +72,16 @@ func (daemon *Daemon) ContainerRm(name string, config *ContainerRmConfig) error
 		}
 		container.LogEvent("destroy")
 
-		if config.RemoveVolume {
-			container.removeMountPoints()
+		for _, m := range container.MountPoints {
+			if _, ok := m.Volume.(*local.Volume); !ok || config.RemoveVolume {
+				if m.Volume != nil {
```

Comment:
I think we want this condition first, it doesn't make sense to check if volume is nil after trying to cast it to local :)